pkg/cluster/internal/providers/provider/provider.go

```diff
@@ -45,4 +45,8 @@ type Provider interface {
 	GetAPIServerInternalEndpoint(cluster string) (string, error)
 	// CollectLogs will populate dir with cluster logs and other debug files
 	CollectLogs(dir string, nodes []nodes.Node) error
+	// SaveImage saves image to dest, as in `docker save`
+	SaveImage(image, dest string) error
```

Comment:
Node providers cannot all provide this anyhow, if I have a `kubectl` + pods based provider, then I want the images to come from the local machine.

Instead we need to cover all the local machine when obtaining images. We can provide a unified UX for that.